scripts/workers/submission_worker.py

```diff
@@ -669,16 +670,39 @@ def process_add_challenge_message(message):
     extract_challenge_data(challenge, phases)
 
 
-def process_submission_callback(body):
+def push_metrics_to_pushgateway(submission_pk, queue_name):
+    try:
+        registry = CollectorRegistry()
+        submissions_unloaded_from_queue = Counter(
+            "submissions_unloaded_from_queue",
+            "Submissions unloaded from queue",
+            ["submission_pk", "queue_name"],
+            registry=registry,
+        )
+        submissions_unloaded_from_queue.labels(submission_pk, queue_name).inc()
+        pushgateway_endpoint = os.environ.get("PUSHGATEWAY_ENDPOINT")
+        job_id = "submission_worker_{}".format(submission_pk)
+        pushadd_to_gateway(pushgateway_endpoint, job=job_id, registry=registry)
+    except Exception as e:
+        logger.exception(
+            "{} Exception when pushing metrics to push gateway: {}".format(
+                SUBMISSION_LOGS_PREFIX, e
+            )
+        )
+
+
+def process_submission_callback(body, queue_name):
     try:
         logger.info(
             "{} [x] Received submission message {}".format(
                 SUBMISSION_LOGS_PREFIX, body
             )
         )
+        submission_pk = json.loads(body)["submission_pk"]
         body = yaml.safe_load(body)
         body = dict((k, int(v)) for k, v in body.items())
         process_submission_message(body)
+        push_metrics_to_pushgateway(submission_pk, queue_name)
```

Comment:
Why are we adding `push_metric` call here? This should be done after message is deleted from the queue. Please make the required changes